### ISSUE TITLE
Unexpected NullabilityMismatch warnings

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#29851]
```

        [Fact]
        public void Overriding_Methods()
        {
            var source = @"
using System.Runtime.CompilerServices;
public abstract class A
{
    [NonNullTypes(false)]
    public abstract System.Action<string> Oblivious1(System.Action<string> x);
    [return: NonNullTypes(false)]
    public abstract System.Action<string> Oblivious2([NonNullTypes(false)] System.Action<string> x);
    public abstract System.Action<string> M3(System.Action<string> x);
    public abstract System.Action<string> M4(System.Action<string> x);
    public abstract System.Action<string>? M5(System.Action<string>? x);
}

public class B1 : A
{
    public override System.Action<string?> Oblivious1(System.Action<string?> x) => throw null;
    public override System.Action<string?> Oblivious2(System.Action<string?> x) => throw null; // warn 3 and 4 // PROTOTYPE(NullableReferenceTypes): Should not warn
    public override System.Action<string?> M3(System.Action<string?> x) => throw null; // warn 5 and 6
    public override System.Action<string?> M4(System.Action<string?> x) => throw null; // warn 7 and 8
    public override System.Action<string?> M5(System.Action<string?> x) => throw null; // warn 9 and 10
}

public class B2 : A
{
    [return: NonNullTypes(false)]
    public override System.Action<string> Oblivious1([NonNullTypes(false)] System.Action<string> x) => throw null;
    public override System.Action<string> Oblivious2(System.Action<string> x) => throw null;
    [NonNullTypes(false)]
    public override System.Action<string> M3(System.Action<string> x) => throw null;
    [return: NonNullTypes(false)]
    public override System.Action<string> M4([NonNullTypes(false)] System.Action<string> x) => throw null;
    [NonNullTypes(false)]
    public override System.Action<string> M5(System.Action<string> x) => throw null;
}
";
            var compilation = CreateCompilation(new[] { source, NonNullTypesTrue, NonNullTypesAttributesDefinition });

            compilation.VerifyDiagnostics(
                // (7,14): error CS0592: Attribute 'NonNullTypes' is not valid on this declaration type. It is only valid on 'module, class, struct, enum, constructor, method, property, indexer, field, event, interface, delegate' declarations.
                //     [return: NonNullTypes(false)]
                Diagnostic(ErrorCode.ERR_AttributeOnBadSymbolType, "NonNullTypes").WithArguments("NonNullTypes", "module, class, struct, enum, constructor, method, property, indexer, field, event, interface, delegate").WithLocation(7, 14),
                // (8,55): error CS0592: Attribute 'NonNullTypes' is not valid on this declaration type. It is only valid on 'module, class, struct, enum, constructor, method, property, indexer, field, event, interface, delegate' declarations.
                //     public abstract System.Action<string> Oblivious2([NonNullTypes(false)] System.Action<string> x);
                Diagnostic(ErrorCode.ERR_AttributeOnBadSymbolType, "NonNullTypes").WithArguments("NonNullTypes", "module, class, struct, enum, constructor, method, property, indexer, field, event, interface, delegate").WithLocation(8, 55),
                // (17,44): warning CS8609: Nullability of reference types in return type doesn't match overridden member.
                //     public override System.Action<string?> Oblivious2(System.Action<string?> x) => throw null; // warn 3 and 4 // PROTOTYPE(NullableReferenceTypes): Should not warn
                Diagnostic(ErrorCode.WRN_NullabilityMismatchInReturnTypeOnOverride, "Oblivious2").WithLocation(17, 44),
                // (17,44): warning CS8610: Nullability of reference types in type of parameter 'x' doesn't match overridden member.
                //     public override System.Action<string?> Oblivious2(System.Action<string?> x) => throw null; // warn 3 and 4 // PROTOTYPE(NullableReferenceTypes): Should not warn
                Diagnostic(ErrorCode.WRN_NullabilityMismatchInParameterTypeOnOverride, "Oblivious2").WithArguments("x").WithLocation(17, 44),
                // (18,44): warning CS8609: Nullability of reference types in return type doesn't match overridden member.
                //     public override System.Action<string?> M3(System.Action<string?> x) => throw null; // warn 5 and 6
                Diagnostic(ErrorCode.WRN_NullabilityMismatchInReturnTypeOnOverride, "M3").WithLocation(18, 44),
                // (18,44): warning CS8610: Nullability of reference types in type of parameter 'x' doesn't match overridden member.
                //     public override System.Action<string?> M3(System.Action<string?> x) => throw null; // warn 5 and 6
                Diagnostic(ErrorCode.WRN_NullabilityMismatchInParameterTypeOnOverride, "M3").WithArguments("x").WithLocation(18, 44),
                // (19,44): warning CS8609: Nullability of reference types in return type doesn't match overridden member.
                //     public override System.Action<string?> M4(System.Action<string?> x) => throw null; // warn 7 and 8
                Diagnostic(ErrorCode.WRN_NullabilityMismatchInReturnTypeOnOverride, "M4").WithLocation(19, 44),
                // (19,44): warning CS8610: Nullability of reference types in type of parameter 'x' doesn't match overridden member.
                //     public override System.Action<string?> M4(System.Action<string?> x) => throw null; // warn 7 and 8
                Diagnostic(ErrorCode.WRN_NullabilityMismatchInParameterTypeOnOverride, "M4").WithArguments("x").WithLocation(19, 44),
                // (20,44): warning CS8609: Nullability of reference types in return type doesn't match overridden member.
                //     public override System.Action<string?> M5(System.Action<string?> x) => throw null; // warn 9 and 10
                Diagnostic(ErrorCode.WRN_NullabilityMismatchInReturnTypeOnOverride, "M5").WithLocation(20, 44),
                // (20,44): warning CS8610: Nullability of reference types in type of parameter 'x' doesn't match overridden member.
                //     public override System.Action<string?> M5(System.Action<string?> x) => throw null; // warn 9 and 10
                Diagnostic(ErrorCode.WRN_NullabilityMismatchInParameterTypeOnOverride, "M5").WithArguments("x").WithLocation(20, 44),
                // (30,14): error CS0592: Attribute 'NonNullTypes' is not valid on this declaration type. It is only valid on 'module, class, struct, enum, constructor, method, property, indexer, field, event, interface, delegate' declarations.
                //     [return: NonNullTypes(false)]
                Diagnostic(ErrorCode.ERR_AttributeOnBadSymbolType, "NonNullTypes").WithArguments("NonNullTypes", "module, class, struct, enum, constructor, method, property, indexer, field, event, interface, delegate").WithLocation(30, 14),
                // (31,47): error CS0592: Attribute 'NonNullTypes' is not valid on this declaration type. It is only valid on 'module, class, struct, enum, constructor, method, property, indexer, field, event, interface, delegate' declarations.
                //     public override System.Action<string> M4([NonNullTypes(false)] System.Action<string> x) => throw null;
                Diagnostic(ErrorCode.ERR_AttributeOnBadSymbolType, "NonNullTypes").WithArguments("NonNullTypes", "module, class, struct, enum, constructor, method, property, indexer, field, event, interface, delegate").WithLocation(31, 47),
                // (25,14): error CS0592: Attribute 'NonNullTypes' is not valid on this declaration type. It is only valid on 'module, class, struct, enum, constructor, method, property, indexer, field, event, interface, delegate' declarations.
                //     [return: NonNullTypes(false)]
                Diagnostic(ErrorCode.ERR_AttributeOnBadSymbolType, "NonNullTypes").WithArguments("NonNullTypes", "module, class, struct, enum, constructor, method, property, indexer, field, event, interface, delegate").WithLocation(25, 14),
                // (26,55): error CS0592: Attribute 'NonNullTypes' is not valid on this declaration type. It is only valid on 'module, class, struct, enum, constructor, method, property, indexer, field, event, interface, delegate' declarations.
                //     public override System.Action<string> Oblivious1([NonNullTypes(false)] System.Action<string> x) => throw null;
                Diagnostic(ErrorCode.ERR_AttributeOnBadSymbolType, "NonNullTypes").WithArguments("NonNullTypes", "module, class, struct, enum, constructor, method, property, indexer, field, event, interface, delegate").WithLocation(26, 55)
                );

            var b1 = compilation.GetTypeByMetadataName("B1");
            verifyMethodMatchesOverridden(expectMatch: false, b1, "Oblivious1");
            verifyMethodMatchesOverridden(expectMatch: false, b1, "Oblivious2");
            verifyMethodMatchesOverridden(expectMatch: false, b1, "M3");
            verifyMethodMatchesOverridden(expectMatch: false, b1, "M4");
            verifyMethodMatchesOverridden(expectMatch: false, b1, "M5");

            var b2 = compilation.GetTypeByMetadataName("B2");
            verifyMethodMatchesOverridden(expectMatch: false, b2, "Oblivious1"); // PROTOTYPE(NullableReferenceTypes): They should match
            verifyMethodMatchesOverridden(expectMatch: true, b2, "Oblivious2"); // PROTOTYPE(NullableReferenceTypes): They should not match
            verifyMethodMatchesOverridden(expectMatch: false, b2, "M3");
            verifyMethodMatchesOverridden(expectMatch: true, b2, "M4"); // PROTOTYPE(NullableReferenceTypes): They should not match
            verifyMethodMatchesOverridden(expectMatch: false, b2, "M5");

            void verifyMethodMatchesOverridden(bool expectMatch, NamedTypeSymbol type, string methodName)
            {
                var member = type.GetMember<MethodSymbol>(methodName);
                Assert.Equal(expectMatch, member.ReturnType.Equals(member.OverriddenMethod.ReturnType, TypeCompareKind.CompareNullableModifiersForReferenceTypes));
                Assert.Equal(expectMatch, member.Parameters.Single().Type.Equals(member.OverriddenMethod.Parameters.Single().Type, TypeCompareKind.CompareNullableModifiersForReferenceTypes));
            }
        }
```

**Comment by jaredpar** [comment]
@AlekseyTs is this issue relevant still? 

I tried doing a rough translation to the new nullable syntax here to check the diagnostics but can't remember the finer points of how the attributes work. For example couldn't remember if applying the attribute to all params / return types was equivalent to `#nullable disable` for the entire method or not.

**Comment by AlekseyTs** [comment]
@jaredpar It looks like the issue was opened to track work for removed PROTOTYPE comments. The unit-test is actually an existing unit-test in NullableReferenceTypesTests.cs, it has been adjusted to the current syntax, but still has comments referencing this issue. I didn't try to figure out if the comments are still relevant. CC @cston.
